### ISSUE TITLE
Hide completed, but only if you've explicitly checked them today

### ISSUE THREAD

**Issue by akfaew** [iSoron/uhabits#565]
I track around 25 habits with the app, and sometimes struggle with organizing them.

I have some physiotherapy exercises that I intend to do 5 times in 7 days at a minimum to have peace of mind, but of course it's better to do them 7 times in 7 days.

I'd also like to shave at least once every three days, but naturally it's better to do it every day.

Currently there is a very convenient option: "Hide completed". It would, however, be more convenient to only hide items I've explicitly completed, showing all others, including those with an "automatic done".

And thank you for making the app @iSoron, it's been remarkably helpful.

**Comment by NellyWhadsDev** [comment]
The current work flow is based on "at least N days in M days" to consider a habit "complete". The proposal seems to be to modify this to be "at least N days up to M days in P days".

This would fit a user story along the lines of "I'd like to shave at least 3 times a week, but 5 times is when I consider my habit schedule met."

This would however introduce another degree of freedom to the "completion" metric. The above example, with 4 days of shaving, would be a "partially complete" habit, since the hard target is 5 days, and the soft target is 3 days. From what I've seen so far in code-land, neither the UI or the score calculation is quite ready for this.

As cop-out-ish as it seems, the best way to accomplish this right now would probably be with 2 separate habits :(

**Comment by hiqua** [comment]
I interpreted the proposal differently, and indeed, I was looking for this issue before creating my own describing what I think @akfaew tried to describe.

Basically there's no completion logic change, but only a new UI option, which is to add a 3rd option to the already existing:

1) Hide archived
2) Hide completed

to end up with something like:

1) Hide archived
2) Hide completed
3) Hide ticked today

(note that here, 3) implies 2) so it's not ideal).

When you have option 3) activated, you would only see the activities you can still tick.


By default, every day, I just want to see what's left to tick. I have a preference for uncompleted habits of course, but if I can tick completed habits it's also good.

To summarize, it's just one more UI filter that does not collide with any user that would not use it. I think there's still the question of how to make 2) and 3) work together nicely though.

**Comment by akfaew** [comment]
Yes, that's what I meant. Sorry if I didn't write it clearly enough. Thank you @hiqua.

**Comment by kkelleyjr** [comment]
I think this is a much needed feature and would be massively helpful. I think this feature would also have to include an additional completion option, x for not done, ✔️ for done, and / for intentionally not doing. With this adjustment we are able to get to a empty screen for 💯 for the day.

**Comment by KristianTashkov** [comment]
@iSoron the comment up top is another vote for "Hide Entered" filter that I wanted to introduce together with the explicit fail.

**Comment by kkelleyjr** [comment]
I later found that there is a request for setting explicitly the days to do the habit and only show them on the days, that would help out this story and not require a new state.

**Comment by iSoron** [comment]
I consider this fixed by #660.
